### PR TITLE
More Survival Tools: Series 2

### PR DESCRIPTION
**Pull request by chaosvolt**
Now that the mod is a thing, opted to release another series of changes
to it.

Major Fixes:
1. Fixed typo in modinfo description, and added mention of recipe
additions and changes.
2. Updated modded Wilderness challenge to be consistent with current
unmodded version.
3. Recipes involving pre-existing items that presently lack recipes in
the main game have been given the "overrides" feature, to idiot-proof
myself against anything breaking should a recipe for said items be
introduced in the main game later.

Recipe Additions and Changes:
1. Allowed large stomachs to be used for curdling milk.
2. Added low-quality hammer quality to recipe for cheese-making, since
you'd be pressing whey from curds.
3. Allowed thyme oil to be used for natural-made bandages.
4. Added thyme stalk to Italian seasoning crafting requirements.
5. Increased crafting recipe of copper axes from 3 survival to 4.
6. Allowed use of fur pelts for crafting leather pouches, easier than
making a fur equivalent.
7. Updated modded stone shovel and axe recipes to allow sharp rocks.
8. Added modded stone adze recipe allowing resinous cords, yarn, duct
tape, or withered plants.
9. Various custom reactions involving stone now allow sharp rocks in
addition to normal rocks.
10. Added recipe for sharp rocks. Since their primary purpose is having a
butchering quality even lower than the trowel, it seems feasible enough
to have them easily available for now.

Item Additions and Changes:
1. Fur leggings, as the non-mutant equivalent to fetlock furs. If it's
good enough for Ötzi, it's good enough for you.
2. Description of curdling/curdled milk no longer specifically mentions
vegetable rennet, due to the possibility of having used animal rennet.
3. Aluminum foil and...tinfoil hats!

Tileset Changes:
1. MShock24Modded tileset has been updated to current standards, and officially branched off with the addition of sprites for the glass macuahuitl and fetlock furs, plus additional sprites for most of the mod's content that I didn't originally make new sprites for. When I inevitably find the few items that don't yet have a tile, I intend to edit my own sprites for them instead of stealing from MShock again. o3o
2. Tile config should be working now, implemented Bevapdin's "Fix walls MShockTilesetModded" fixes here, did it the hard way due to the sprite additions to this PR.
3. Addition sprite assignments/creations for various things that lacked sprites.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Fuck. I just tested something. KA101's additions to some terrain loading files means game will break with starting/loading a file that doesn't have my mod loaded...bloody hell. ._.

File: data/mods/More_Survival_Tools/recipes.json [hunk 1]
```diff
@@ -1040,9 +1090,68 @@
     {"id":"HAMMER","level":1,"amount":1}
   ],
   "components": [
-    [ [ "rock", 5 ] ],
+    [ [ "rock", 5 ], [ "ceramic_shard", 1 ], [ "sharp_rock", 1 ] ],
```
Comment (Coolthulhu) [inline data/mods/More_Survival_Tools/recipes.json]: Why only 1 shard/sharp rock but 5 regular rocks?

Also, at 3 super/bone glue it is not worth the effort. 1 would be fine.

Comment (chaosvolt) [inline data/mods/More_Survival_Tools/recipes.json]: That would be a copy-paste fail on my part it seems,

Also, 3 superglue is actually what the unmodded recipe for the glass macuahuitl demands, so I might tone that down to 1 super glue. Should I retain the idea of it requiring more bone glue, or not? o3o

**Comment by chaosvolt** [comment]
I'm also possibly having some problem with the tileset all of a sudden, based off one of the recent compiled experimentals at least.

Wall tiles are using the buggy fallback instead of the tiles they should be using.

Spawning wall tiles in the map editor has them appear normal, except any such tiles in contact with an existing buggy tile become buggy as well. O.o

**Comment by BevapDin** [comment]
Your tileset still has the _v and _h versions of the wall terrains (#11895).

**Comment by chaosvolt** [comment]
Oh derp, that explains it. >.o

**Comment by BevapDin** [comment]
> Should I add that to this current PR, then?

Yes, do so and make sure it shows the correct walls. The blue wall appears as blue rectangle (even when connected to another wall), but the other colored walls appear as colored double lines (like in your picture), I don't known whether that is correct.

**Comment by chaosvolt** [comment]
Now if I can figure out the easiest way to do so...fine time to be out on my laptop. @_@

**Comment by KA101** [comment]
FWIW, without that change it was fucking up when I tried to load it anyway.  If someone has a fix, feel free.

**Comment by chaosvolt** [comment]
Which change, KA101?

**Comment by KA101** [comment]
The ones you're mentioning.

**Comment by chaosvolt** [comment]
Ah, so hopefully this PR fixes that. I'm guessing you specifically mean the t_leanto load error that occurs on worlds not using the mod?

**Comment by KA101** [comment]
Yes.  Apparently I caused the error people are reporting, and my apologies therefor, since I wasn't sure how else to fix the error I'd been getting at the time.  (And this error didn't appear: I'm guessing because it only hits the first time one fires up the game, so reloading once I'd run a test managed to get around it.)

**Comment by chaosvolt** [comment]
Ah, right. But yeah, it seems a dummy version of t_leanto in the default terrain file fixes that without being noticeable.

Unless you want to mainline it instead? ^^"

**Comment by KA101** [comment]
>Unless you want to mainline it instead?

I don't recall being opposed to that.

**Comment by chaosvolt** [comment]
Hmm, I might add that to this PR when I'm back on the desktop then.

Any other content that might warrant a mainline?

Only other issue is it'd still have the issue of transparency issues on the unmodded MShock tileset and presumably the others as well. =w=

**Comment by KA101** [comment]
> I might add that to this PR

*Separate PR* please.

**Comment by chaosvolt** [comment]
Very well then, shall remain focused on the mod for now. ^^

Now hopefully this is the last of the bugs for now.

**Comment by chaosvolt** [comment]
Also, I just derped and realized that there seems to be no recipe for knapping a sharp rock from a normal rock.

Any PRs planning to add that as mainline, or should I ponder a reaction for it? o3o

**Comment by KA101** [comment]
Knapping has been discussed repeatedly and absent proper geology/mineralogy support (not quite DF-level but that general idea) we can't justify using one rock to sharpen another.  So modpack is an uphill push,and mainline is No.

**Comment by chaosvolt** [comment]
Ah, so I see. I like the idea of a sharpened rock being just barely good enough to butcher with being readily available though.

After all, if I recall the sharp rock's butcher quality is even worse than a bloody trowel. So given that it is having no cut quality, it seems feasible to believe that knapping a sharp rock would basically be pebble-core technology. Not just more primitive than the existing stone knife, but more primitive than most stone tools in general. Oldowan industry at the latest.

**Comment by chaosvolt** [comment]
Also, since I'm pondering what else might be worth adding, I'll list all content that I removed before PR'ing due to deciding the Classical/Medieval gear doesn't quite fit with the rest of the content:

Heater shields, kite shields, round shields, hoplons, scuta, bucklers, riot shields, wooden shields, Corinthian helmets, bell cuirasses, bronze greaves, lorica segmentata, galeae, nasal helmets, tunics, mail armor (now obsolete anyway, though my version implied being paired with a gambeson), baldrics, improvised sleds (deployable item like a folding bike, obsolete due to the mechanics-skill-0 travois being about the same), xiphoses, dories, bronze, arming swords, and scimitars.

**Comment by chaosvolt** [comment]
Huh. Just checked the latest compiled experimental, someone already fixed the t_leanto load error? O.o

**Comment by narc0tiq** [comment]
Yes, I did. I was impatient.

**Comment by chaosvolt** [comment]
Ah, thank you. I didn't see any terrain tweaks, so what was the fix? owo

**Comment by narc0tiq** [comment]
#11958

**Comment by chaosvolt** [comment]
Ah I see. So long as that tweak KA101 added has proven unnecessary. If this PR gets through, I would hope adding a dummy version of t_leanto won't break anything?

That way if KA101 does re-instate that C++ tweak it won't cause the bug to re-surface. >.o

**Comment by chaosvolt** [comment]
Oh wait, one small problem here. Your change to it also means lean-tos no longer count as shelter to sleep in? O.o

**Comment by KA101** [comment]
Rest assured KA101 won't reinstate the tweak: that came from him trying to fix something without time/energy.

**Comment by chaosvolt** [comment]
Hmm. That would mean that the lean-to no longer functions as shelter though, wouldn't it?

My addition of a dummy version in the default terrain file would presumably fix things as well. @_@

**Comment by narc0tiq** [comment]
It will work exactly as much either way -- I simply removed a variable declaration and initialization. There was never any behaviour attached to the variable.

You might be thinking of #11959, but the `t_improvised_shelter` variable does have behaviour attached: <https://github.com/CleverRaven/Cataclysm-DDA/blob/5976102/src/player.cpp#L830> and <https://github.com/CleverRaven/Cataclysm-DDA/blob/5976102/src/player.cpp#L11561>. Initializing the variable makes it work for the correct terrain instead of for the default null terrain.

**Comment by chaosvolt** [comment]
Huh, I see. So long as the lean-to still functions for the sake of keeping rain off of you, then I'm fine with it as it is. o3o

**Comment by chaosvolt** [comment]
Pondering what food items might warrant using foil as a container item. Additionally pondering other cheap cookware, such as maybe non-electronic mess kits? One would assume they'd offer at least level 1 cooking and boiling, in much the same way that the frying pan has less cooking utility in exchange for less volume.

**Comment by DavidKeaton** [comment]
Foil could be used to slow heat degradation

**Comment by chaosvolt** [comment]
Ooh, that would be an interesting use for it, depends on if a flag exists for that.

**Comment by Zireael07** [comment]
I like improvised mess kits and foil use as keeping food hot longer.

**Comment by DavidKeaton** [comment]
@chaosvolt You may have to implement said flag. The flag checks could be done during food processing, and you could make the time adjustments there.

**Comment by chaosvolt** [comment]
That could be awkward to implement. Also, not improvised mess kits, but more like existing military mess kits. An old-school example: http://en.wikipedia.org/wiki/File:US_Army_mess_kits_-_IMG_1506.JPG

**Comment by DavidKeaton** [comment]
@chaosvolt I still have some of the forks that came from those things. Also, those cool little p42s (or whatever those can openers are called) ftw.

**Comment by chaosvolt** [comment]
Be lucky that easy-open cans are apparently ubiquitous in the Cataclysm future. P38 can openers would be a neat addition if we needed to open cans though. o3o

Will have to see what other low-tech cooking items already exist, there might already be something like that in the. Will also have to look at the frying pan to see what qualities it has, since I figure a mess tin should be balanced by having even lower qualities for further reduced bulk. But if the qualities are low enough that further applications of the nerf stick would make the resulting item no better than just carrying a skewer and soda can...

**Comment by chaosvolt** [comment]
There we go. I think I'm officially ready to leave this to KA101.

**Comment by chaosvolt** [comment]
Hmm. The commit validation is automated, right? No change yet, so I'm wondering why it's been almost 24 hours since updates 11 and 12 and they STILL are showing as unvalidated. >w>

**Comment by Coolthulhu** [comment]
Pretty sure they're automatic, but they sometimes break. In case there are many unscanned commits pushed at once, Jently will only scan the last one.

**Comment by chaosvolt** [comment]
Ah derp. That would explain why updates 11 and 12 haven't been scanned, since I had to commit them one right after the other due to using the online editor at that time. ._.

**Comment by chaosvolt** [comment]
Oh. OH. Right, reverts don't actually remove the commits, just counter them with new commits. Because that would TOTALLY solve the "Jently isn't doing anything" problem.

Scheisse. ._.

**Comment by narc0tiq** [comment]
I just kicked Jently with the general IT fix (turn it off, then on again). Seems to work better now.

**Comment by chaosvolt** [comment]
Huh. It did however do as Coolthulhu said it'd do and skipped straight to the re-commit of updates 11-13. Oh well. o3o

**Comment by narc0tiq** [comment]
Well, uh, that contains everything before it, so... duh?

**Comment by chaosvolt** [comment]
Yus. Nothing of value was skipped, in that case. o3o

**Comment by chaosvolt** [comment]
And sorry about that. I think next time I start an update, I'll keep tileset updates on a separate PR. ._.

**Comment by Coolthulhu** [comment]
Merged.

For the future:
* Hand axe is really good. As a weapon, it is better than stone macuahuitl and it's supposed to be a tool and not a weapon. As a tool it is also great.
* Hard cheese tastes just as well as the regular one. Isn't it supposed to be more of a survival thing?
* Fur leggings cost 8 fur, which is quite a lot for just 40 warmth. For 15 fur you could have a fur cloak that gives 60 on everything except hands, feet and mouth
* You could get rid of that COOK 1 on aluminum foil. It doesn't fit on skewers, but neither it does fit here
* Copper axe should probably use fabrication, not survival
* You probably don't need a rock or a hammer to press whey out of milk. A rag or something like that would be useful, though - you need to press the milk through something

**Comment by chaosvolt** [comment]
Hmm.

For the hand axe, will need to tone down the damage it seems. Probably should be not much better than the sharp rock, damage wise.

Hard cheese probably should, but logically it would depend on the quality of homemade cheese. Most commercially available hard cheeses are delicious in my opinion, but those obviously aren't made by lone individuals innawoods. o3o

I think the reason behind the leggings using what should be the same amount as fetlock furs is the fact that toning down the number of pelt scraps would cause some weird-looking ratios when compared to either retaining the need for 2 full pelts, or toning THAT down to 1. Doing the latter would require I add a cutting quality for it to still make sense, given the assumption that fur leggings are still separate sections.

As for aluminum foil wraps, I wanted there to some function to it besides making tinfoil hats, which of course became pointless when someone else PR'd a version of that concept that was actually functional rather than cosmetic as mine was. At this point it might warrant pulling aluminum foil wraps entirely. =w=

And likely should, I probably based copper axe recipe off stone axe. The skill requirements for copperworking likely depend on whether heating copper over a campfire and working it would be chalcolithic enough to warrant adding any survival skill, versus treating it as more like entry-level metalsmithing knowledge.

Ooh, cheesecloth would make more sense too. o3o

**Pull request merged**